language/resource-api/README.md

```diff
@@ -69,16 +70,16 @@ The `Puppet::ResourceApi.register_type(options)` function takes the following ke
 * `autorequire`, `autobefore`, `autosubscribe`, and `autonotify`: a hash mapping resource types to titles. The titles must either be constants, or, if the value starts with a dollar sign, a reference to the value of an attribute. If the specified resources exist in the catalog, Puppet will create the relationsships requested here.
 * `features`: a list of API feature names, specifying which optional parts of this spec the provider supports. Currently defined features: `canonicalize`, `simple_get_filter`, and `supports_noop`. See below for details.
 
-For autoloading work, this code needs to go into `lib/puppet/type/<name>.rb` in your module.
+For autoloading to work, this code needs to go into `lib/puppet/type/<name>.rb` in your module.
 
 ###  Composite Namevars ("title_patterns")
 
-Each resource being managed must be identified by a unique title. Usually this is fairly straightforward and a single attribute can be used to act as an identifier. Sometimes though, you need a composite of two attributes to uniquely identify the resource you want to manage.
+Each resource being managed must be identified by a unique title. Usually this is fairly straightforward and a single attribute can be used to act as an identifier. Sometimes though, you need a composite of two or more attributes to uniquely identify the resource you want to manage.
 
-If multiple attributes are defined with the `namevar` behaviour, the type SHOULD specify `title_patterns` that will tell Resource API how to get at the attributes from the title. If `title_patterns` is not specified a default pattern is applied, and matches against the first declared `namevar`.
+If multiple attributes are defined with the `namevar` behaviour, the type SHOULD specify `title_patterns` that will tell Resource API how to get at the attributes from the title. If `title_patterns` is not specified a default pattern is applied, and matches the entire title against the first declared `namevar`.
+> Note: Not supplying a title_pattern that assigns values to all namevars will severely restrict the usability of the provider.
```

Comment:
So what should you instead do?

Reply:
As the text says: if `title_patterns` is not specified a default pattern is applied, and matches the entire title against the first declared `namevar`. This will mean that in puppet code all other namevars need to be specified as attributes in the body of the resource.

Reply:
I still feel line 80 is dangling - comes across as a threat as I don't really understand what it is telling me. I understand line 79, it is telling me what I should/must do.

Reply:
Ah... I think it is saying "A supplied title pattern MUST assign values to all namevars, as not doing this will severely restrict the usability of the provider."
Did I get it?

Reply:
Sorry to harp on this, but since I don't know how it actually works and would like to know... I keep coming back to this :-)
When you say "As the text says: if title_patterns is not specified a default pattern is applied, and matches the entire title against the first declared namevar. This will mean that in puppet code all other namevars need to be specified as attributes in the body of the resource." - is that in any way recommended way of doing it? Are people doing it? If so, you may want to include something about this - "In legacy code this may have been implemented as <some variant of what you wrote> ... This is discouraged (deprecated even?) as it strongly restricts the usability of the provider"

Reply:
I'm trying to toe the line between "it's not required" and "if you try and it breaks, you may keep both pieces".

I should just stamp down on it and make not having a `title_pattern` that assigns all namevars an error.

Reply:
Yes, and then change the `SHOULD` to `MUST` - we take no responsibility, nor do we tell you what will happen - it is simply undefined and W.R.O.N.G.

Reply:
Yeah, I'll do that - even if it's possibly not something we can actively check on the backend. :man_shrugging: